core/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -919,6 +921,23 @@ protected int estimatedSizeInBytes() {
             return (id().length() + type().length()) * 2 + source().length() + 12;
         }
 
+        /**
+         * Returns a positive timestamp if the ID of this document is auto-generated by elasticsearch.
+         * if this property is non-negative indexing code might optimize the addition of this document
+         * due to it's append only nature.
+         */
+        public long getAutoGeneratedIdTimestamp() {
+            return autoGeneratedIdTimestamp;
+        }
+
+        /**
+         * Returns <code>true</code> iff this index requests has been retried on the coordinating node and can therefor be delivered
```

Comment:
I don't think it's important since we handle it in the engine right even if it's not set